src/Microsoft.AspNet.OData/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -225,6 +225,31 @@ public static Uri GetNextPageLink(this HttpRequestMessage request, int pageSize)
             return GetNextPageHelper.GetNextPageLink(requestUri, request.GetQueryNameValuePairs(), pageSize);
         }
 
+        /// <summary>
+        /// Creates a link for the next page of results; To be used as the value of @odata.nextLink.
+        /// </summary>
+        /// <param name="request">The request on which to base the next page link.</param>
+        /// <param name="pageSize">The number of results allowed per page.</param>
+        /// <param name="instance">The instance based on which the skiptoken value is generated. </param>
+        /// <param name="objToSkipTokenValue">Function that extracts out the skiptoken value from the object</param>
+        /// <returns>A next page link.</returns>
+        public static Uri GetNextPageLink(this HttpRequestMessage request, int pageSize, object instance, Func<object, string> objToSkipTokenValue)
+        {
+            if (request == null || request.RequestUri == null)
+            {
+                throw Error.ArgumentNull("request");
+            }
+
+            Uri requestUri = request.RequestUri;
+
+            if (!requestUri.IsAbsoluteUri)
+            {
+                throw Error.ArgumentUriNotAbsolute("request", requestUri);
+            }
```

Comment:
why do we require the request to be absolute?

Reply:
The URL is opaque, but can either be absolute or relative to the current context. There are scenarios where a front end routes requests to backends that are not aware of the front end so they always return relative URLs to make sure that requests continue to go through the front-end.

---
In reply to: [248866026](https://github.com/OData/WebApi/pull/1724#discussion_r248866026) [](ancestors = 248866026,248331626)